### PR TITLE
Use cell defined constants for various stuff

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
This re writes most cell power usage cases with 2 defines `STANDARD_CELL_CHARGE`(Joules) & `STANDARD_CELL_RATE`(Watts) so changing cell capacity values in the future won't cause discrepancies.

## Changelog
:cl:
code: most cell power usages are scaled with defined constants to help adapt to future changes
/:cl:

### CODE REVIEW

**Comment by SyncIt21** [comment]
@Pickle-Coding review & what not

**Comment by Pickle-Coding** [comment]
> @Pickle-Coding review & what not.
> 
> 
> 
> The only problems we have are with electrolyser & canister shielding. When using from the grid its fine but when using from the cell we scale it by 1000x. 
> 
> 
> 
> This will break if we change standard cell charge in the future so should i remove that scaling entirely or do we have a better solution?

What is the power consumption without the scalingI might properly review this tomorrow if I have time.

For now I'd prefer if the energy usage was consistent, so pick whether you want to apply cell or grid power usage to both.

File: code/game/machinery/suit_storage_unit.dm [hunk 1]
```diff
@@ -54,10 +57,8 @@
 	var/message_cooldown
 	/// How long it takes to break out of the SSU.
 	var/breakout_time = 30 SECONDS
-	/// Power contributed by this machine to charge the mod suits cell without any capacitors
-	var/base_charge_rate = 200 KILO WATTS
```
Comment (MrMelbert) [inline code/game/machinery/suit_storage_unit.dm:58]: Should leave this as a var so subtypes can edit it and admins can VV it

Comment (SyncIt21) [reply code/game/machinery/suit_storage_unit.dm:58]: But we don't have enough subtypes for suit storage. But yeah I'll leave it for admins then

**Comment by SyncIt21** [comment]
> What is the power consumption without the scaling? I might properly review this tomorrow if I have time.
> 
> For now I'd prefer if the energy usage is consistent, so pick whether you want to apply cell or grid power usage to both.


This should explain how power is used between the 2. Now both use the same energy without scaling. https://github.com/tgstation/tgstation/pull/82594/commits/79f983a939d1f5b6128677a5f50dd32880fb0f9e

File: code/__DEFINES/lights.dm [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ///How much power emergency lights will consume per tick
-#define LIGHT_EMERGENCY_POWER_USE (0.1 KILO WATTS)
+#define LIGHT_EMERGENCY_POWER_USE (0.0001 * STANDARD_CELL_RATE)
```
Comment (jlsnow301) [inline code/__DEFINES/lights.dm:2]: i feel like the defines are just making things harder to read

Comment (SyncIt21) [reply code/__DEFINES/lights.dm:2]: True, but we need standards and  not magic numbers.  We have material sheet defines for mats and standard cell defines for power 

Also Hi jlsnow :) hope your doing well

File: code/__DEFINES/power.dm [hunk 3]
```diff
@@ -19,8 +19,12 @@
...
 #define JOULES * JOULE
 
-///The amount of energy, in joules, a standard powercell has.
-#define STANDARD_CELL_CHARGE (1 MEGA JOULES) // 1 MJ.
+///The capacity of a standard power cell
+#define STANDARD_CELL_VALUE (1 MEGA)
+	///The amount of energy, in joules, a standard powercell has.
+	#define STANDARD_CELL_CHARGE (STANDARD_CELL_VALUE JOULES) // 1 MJ.
+	///The amount of power, in watts, a standard powercell can give.
+	#define STANDARD_CELL_RATE (STANDARD_CELL_VALUE WATTS) // 1 MW.
```
Comment (Pickle-Coding) [inline code/__DEFINES/power.dm:27]: I think the documentation on this one could be improved. The description implies that there is some limit on the power usage of a cell, which is not the case. Also why is it indented?

**Comment by Pickle-Coding** [comment]
Seems mostly fine. Not sure why some variables are getting replaced with defines though.

File: code/__DEFINES/power.dm (hunk 3 above)
Comment (SyncIt21) [reply code/__DEFINES/power.dm:27]: Indendation is simply for readibility. It shows us that this define depends on(or you could say a subtype) of the above define.

Here's another example 
https://github.com/tgstation/tgstation/blob/1c81752af4cfecce02114809895a3cf15b962ec3/code/__DEFINES/dcs/signals/signals_material_container.dm#L6-#L7

The return value is related to the signal so to show that relationship we indent it.

Also yes you are right about the documentation, I don't know how else to better write it so I'm open to suggestions

**Comment by SyncIt21** [comment]
> Seems mostly fine. Not sure why some variables are getting replaced with defines though.

I di that in places where the variables are read only(their value never changes). It can't be called a variable at that point cause it's not "varying" so better define it as a constant and save memory

**Pull request merged**